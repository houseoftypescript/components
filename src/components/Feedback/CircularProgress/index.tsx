import React from 'react';
import styled from 'styled-components';

const CircularProgressContainer = styled.svg`
  transform: rotate(-90deg);

  circle {
    transition: all 100ms linear;
  }
`;

const CircularProgress: React.FC<{ value: number }> = ({ value = 0 }) => {
  return (
    <CircularProgressContainer width="100" height="100" viewBox="0 0 100 100">
      <circle
        cx="50"
        cy="50"
        r="45"
        strokeWidth="10"
        fill="transparent"
        stroke="#d1d5db"
      />
      <circle
        cx="50"
        cy="50"
        r="45"
        fill="transparent"
        stroke="#ef4444"
        strokeWidth="10"
        strokeDashoffset={100 - value}
        strokeDasharray={100}
        pathLength={100}
      />
    </CircularProgressContainer>
  );
};

export default CircularProgress;
