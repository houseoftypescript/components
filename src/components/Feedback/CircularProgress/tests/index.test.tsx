import { render } from '@testing-library/react';
import React from 'react';
import CircularProgress from '../index';

describe('CircularProgress', () => {
  it('to match snapshot', () => {
    const { container } = render(<CircularProgress value={50} />);
    expect(container).toMatchSnapshot();
  });
});
