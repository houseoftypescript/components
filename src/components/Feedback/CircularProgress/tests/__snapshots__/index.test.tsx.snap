// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress to match snapshot 1`] = `
<div>
  <svg
    class="sc-bcXHqe bBFQQB"
    height="100"
    viewBox="0 0 100 100"
    width="100"
  >
    <circle
      cx="50"
      cy="50"
      fill="transparent"
      r="45"
      stroke="#d1d5db"
      stroke-width="10"
    />
    <circle
      cx="50"
      cy="50"
      fill="transparent"
      pathLength="100"
      r="45"
      stroke="#ef4444"
      stroke-dasharray="100"
      stroke-dashoffset="50"
      stroke-width="10"
    />
  </svg>
</div>
`;
